warehouse/query-core/src/main/java/datawave/query/tables/shard/IndexStatsQueryLogic.java

```diff
@@ -156,10 +156,11 @@ public List<FieldStat> getStat(Set<String> fields, Set<String> dataTypes, Sorted
                     dataSource = simpleScanner;
                     scanner = simpleScanner;
                 } else {
-                    BatchScanner bScanner = ScannerHelper.createBatchScanner(con, table, auths, fields.size());
-                    bScanner.setRanges(buildRanges(fields));
-                    scanner = bScanner;
-                    dataSource = bScanner;
+                    try (BatchScanner bScanner = ScannerHelper.createBatchScanner(con, table, auths, fields.size())) {
```

Comment:
This is another case where try-with-resources would close the batch scanner before we're done with it.

Reply:
refactored this -- think it should be better.

Reply:
reverted.